src/PolarCircleResolution.js

```diff
@@ -0,0 +1,119 @@
+import SolarTime from './SolarTime';
+import { dateByAddingDays } from './DateUtils';
+
+export const PolarCircleResolution = {
+  aqrabBalad: 'aqrab-balad',
+  aqrabYaum: 'aqrab-yaum',
+  unresolved: 'unresolved'
+};
+
+const LATITUDE_VARIATION_STEP = 0.5; // Degrees to add/remove at each resolution step
+const UNSAFE_LATITUDE = 65; // Based on https://en.wikipedia.org/wiki/Midnight_sun
+
+const isValidSolarTime = (solarTime) => {
+  if (solarTime) {
+    return solarTime && !isNaN(solarTime.sunrise) && !isNaN(solarTime.sunset);
+  }
+  return false;
+}
+
+const aqrabYaumResolver = (coordinates, date, daysAdded) => {
+  if (daysAdded <= Math.ceil(365 / 2)) {
```

Comment:
As a stylistic thing, I'm a fan of early return as to avoid nesting. For this scenario we could put it as

```
if (daysAdded > Math.ceil(365 / 2)) {
    return null;
}
```